packages/protocol/lib/compatibility/ast-layout.ts

```diff
@@ -97,14 +102,36 @@ const generateLayoutCompatibilityReport = (oldLayout: StorageLayoutInfo, newLayo
   }
 }
 
-const compareStructDefinitions = (oldType: TypeInfo, newType: TypeInfo) => {
+const compareStructDefinitions = (oldType: TypeInfo, newType: TypeInfo, structExpandable: boolean) => {
   if (oldType.kind !== 'struct') {
     return {
       same: false,
       errors: [`${newType.label} wasn't a struct type, now is`]
     }
   }
 
+  if (structExpandable && oldType.members.length < newType.members.length) {
+    const expandableErrors = oldType.members.map((oldMember, i) => {
+      const newMember = newType.members[i]
+
+      if (oldMember.label !== newMember.label && `deprecated_${oldMember.label}` !== newMember.label) {
+        return `struct ${newType.label} had ${oldMember.label} in slot ${i}, now has ${newMember.label}`
+      }
+  
+      if (oldMember.type !== newMember.type) {
+        return `struct ${newType.label}'s member ${newMember.label} changed type from ${oldMember.type} to ${newMember.type}`
+      }
+    }).filter(error => error)
+
+    if (expandableErrors.length === 0) {
+      return {
+        same: true,
```

Comment:
Naming nit: guess this should be renamed to something like `compatible`, since now structs can be different but still compatible.